core/src/retransmit_stage.rs

```diff
@@ -47,21 +47,22 @@ fn retransmit(
         GROW_LAYER_CAPACITY,
     );
     for b in &dq {
-        if b.read().unwrap().should_forward() {
-            ClusterInfo::retransmit_to(&cluster_info, &neighbors, &copy_for_neighbors(b), sock)?;
+        if b.read().unwrap().meta.forward {
+            ClusterInfo::retransmit_to(&cluster_info, &neighbors, &mark_forwarded(b), sock)?;
+            ClusterInfo::retransmit_to(&cluster_info, &children, b, sock)?;
+        } else {
+            ClusterInfo::retransmit_to(&cluster_info, &children, &mark_forwarded(b), sock)?;
         }
-        // Always send blobs to children
-        ClusterInfo::retransmit_to(&cluster_info, &children, b, sock)?;
     }
     Ok(())
 }
 
-/// Modifies a blob for neighbors nodes
+/// Copies and marks a blob as "forwarded"
 #[inline]
-fn copy_for_neighbors(b: &SharedBlob) -> SharedBlob {
+fn mark_forwarded(b: &SharedBlob) -> SharedBlob {
     let mut blob = b.read().unwrap().clone();
```

Comment:
ew!

Reply:
Can't avoid the clone since retransmit_to will par_iter over a readlock on this blob. Couldn't go through with your "hold a write lock" suggestion :(

Reply:
^ Nvm this might be doable, messing around with it now